bin/load_db_scxa_dimred.sh

```diff
@@ -12,55 +12,43 @@ source $scriptDir/db_scxa_common.sh
 
 dbConnection=${dbConnection:-$1}
 EXP_ID=${EXP_ID:-$2}
-EXPERIMENT_DIMRED_PATH=${EXPERIMENT_DIMRED_PATH:-$3}
-SCRATCH_DIR=${SCRATCH_DIR:-"$EXPERIMENT_DIMRED_PATH"}
-TSNE_PREFIX=${TSNE_PREFIX:-"$EXP_ID.tsne_perp_"}
-UMAP_PREFIX=${UMAP_PREFIX:-"$EXP_ID.umap_neigh_"}
-DIMRED_SUFFIX=${DIMRED_SUFFIX:-".tsv"}
+DIMRED_TYPE=${DIMRED_TYPE:-$3}
+DIMRED_FILE_PATH=${DIMRED_FILE_PATH:-$4}
+DIMRED_PARAM_JSON=${DIMRED_PARAM_JSON:-$5}
 
 # Check that necessary environment variables are defined.
-[ ! -z ${dbConnection+x} ] || (echo "Env var dbConnection for the database connection needs to be defined. This includes the database name." && exit 1)
-[ ! -z ${EXP_ID+x} ] || (echo "Env var EXP_ID for the id/accession of the experiment needs to be defined." && exit 1)
-[ ! -z ${EXPERIMENT_DIMRED_PATH+x} ] || (echo "Env var EXPERIMENT_DIMRED_PATH for location of marker genes files for web needs to be defined." && exit 1)
+[ -n ${dbConnection+x} ] || (echo "Env var dbConnection for the database connection needs to be defined. This includes the database name." && exit 1)
+[ -n ${EXP_ID+x} ] || (echo "Env var EXP_ID for the id/accession of the experiment needs to be defined." && exit 1)
+[ -n ${DIMRED_TYPE+x} ] || (echo "Env var DIMRED_TYPE for the dimension reduction type needs to be defined." && exit 1)
+[ -n ${DIMRED_FILE_PATH+x} ] || (echo "Env var DIMRED_FILE_PATH for location of a TSV-format coordinates file for a dimension reduction needs to be defined." && exit 1)
+[ -n ${DIMRED_PARAM_JSON+x} ] || (echo "Env var DIMRED_PARAM_JSON, holding JSON-format parameters for dimension reduction, needs to be defined." && exit 1)
 
 # Check that files are in place.
-[ $(ls -1 $EXPERIMENT_DIMRED_PATH/$TSNE_PREFIX*$DIMRED_SUFFIX | wc -l) -gt 0 ] \
-  || (echo "No tsne tsv files could be found on $EXPERIMENT_DIMRED_PATH" && exit 1)
+if [ ! -s "$DIMRED_FILE_PATH" ]; then
+    echo "No valid file found at $DIMRED_FILE_PATH" && exit 1
+fi
 
 # Check that database connection is valid
 checkDatabaseConnection $dbConnection
 
-
 # Write to the new generic coordinates table
 
 echo "Dimension reductions: Loading data for $EXP_ID (new layout)..."
 rm -f $SCRATCH_DIR/dimredDataToLoad.csv
 
-# Delete table content for current EXP_ID
-echo "coords table: Delete rows for $EXP_ID:"
-echo "DELETE FROM scxa_coords WHERE experiment_accession = '"$EXP_ID"'" | \
-  psql -v ON_ERROR_STOP=1 $dbConnection
+# Insert a new row into the dimension reductions table
+echo "INSERT INTO scxa_dimension_reduction (experiment_accession, method, parameterisation) VALUES ('$EXP_ID', '$DIMRED_TYPE', '$DIMRED_PARAM_JSON');" | psql -v ON_ERROR_STOP=1 $dbConnection
```

Comment:
Won't this fail when attempting to insert a combination that already exists? Previously loaded for the same experiment I mean?

Reply:
I mean given that the delete above is gone. Or are we piggy backing on a delete that will occur outside before calling this script?

Reply:
So, this is due to a strategy change. This script now loads a single dim. red at a time, so I wanted deletion control outside of the script, and called the provided deletion script instead, before loading starts. See https://github.com/ebi-gene-expression-group/atlas-prod/blob/1648812bd228bf784b9ad22e53554c1d2ef95006/exec/import_scxa_experiment.sh#L314